azbfs/datalake_storage_transforms.go

```diff
@@ -0,0 +1,31 @@
+package azbfs
+
+import (
+	"encoding/base64"
+	"strings"
+)
+
+// Converts metadata into a string of format "key1=value1, key2=value2" and Base64 encodes the values.
+func BuildMetadataString(md map[string]string) *string {
```

Comment:
Maps are inherently nullable. I fought this decision on track 2 because it's rather poor UX to have a pointer to something that's already nullable to make it "optional".

Reply:
Made the method private, since maps are nullable I will leave the input as is.